### PR TITLE
New SDK for 2018-06-30 changes and other nit fixes ver2

### PR DESCRIPTION
**Pull request by D1v38om83r**
## Description
1. Python2Packages for azure automation. Added new API specs, re-recorded tests for Automation scenarios. Added a new test for Python2Package scenarios.
API PR: Azure/azure-rest-api-specs#3713

2. Dynamic Group support change
API PR : Azure/azure-rest-api-specs#4165

3. Pre- post support change
API PR : Azure/azure-rest-api-specs#4298

Following on from https://github.com/Azure/azure-sdk-for-net/pull/4916

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by D1v38om83r** [comment]
@dsgouda 
Continuation of PR https://github.com/Azure/azure-sdk-for-net/pull/4971
These are the changes that we recovered from a saved branch. 

They have all the changes minus the ones that you committed in the course of the last PR.

File: src/SDKs/Automation/Management.Automation/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -8,7 +8,7 @@
 [assembly: AssemblyDescription("Provides Microsoft Azure Automation management operations including the ability to create, update and delete runbooks and schedules.")]
 
 [assembly: AssemblyVersion("3.0.1.0")]
-[assembly: AssemblyFileVersion("3.3.0.0")]
+[assembly: AssemblyFileVersion("3.8.0-preview")]
```
Comment (dsgouda) [inline src/SDKs/Automation/Management.Automation/Properties/AssemblyInfo.cs]: Please rename this to `3.8.0.0` (-preview is not allowed in file versioning)

File: src/SDKs/Automation/Management.Automation/Microsoft.Azure.Management.Automation.csproj [hunk 2]
```diff
@@ -6,7 +6,7 @@
   <PropertyGroup>
     <PackageId>Microsoft.Azure.Management.Automation</PackageId>
     <Description>Provides Microsoft Azure Automation management operations including the ability to create, update and delete runbooks and schedules.</Description>
-    <Version>3.4.0-preview</Version>
+    <Version>3.8.0-preview</Version>
     <!-- Remove the Preview description in PackageReleaseNotes once the preview is removed.-->
     <AssemblyTitle>Microsoft Azure Automation Management Library</AssemblyTitle>
```
Comment (dsgouda) [inline src/SDKs/Automation/Management.Automation/Microsoft.Azure.Management.Automation.csproj]: Still missing `PackageReleaseNotes`

**Review by dsgouda — CHANGES_REQUESTED** [review]
All good, just needs a couple of quick fixes

File: src/SDKs/Automation/Management.Automation/Properties/AssemblyInfo.cs (hunk 1 above)
Comment (D1v38om83r) [reply src/SDKs/Automation/Management.Automation/Properties/AssemblyInfo.cs]: done

File: src/SDKs/Automation/Management.Automation/Microsoft.Azure.Management.Automation.csproj (hunk 2 above)
Comment (D1v38om83r) [reply src/SDKs/Automation/Management.Automation/Microsoft.Azure.Management.Automation.csproj]: Added package release notes.

**Comment by D1v38om83r** [comment]
@dsgouda
src\SDKs\_metadata\automation_resource-manager.txt was changed as a part of SDK generation tool. Is it a relevant change?

**Comment by dsgouda** [comment]
Yes this is expected

**Review by dsgouda — APPROVED** [review]
LGTM

**Comment by dsgouda** [comment]
@D1v38om83r appreciate your patience.

**Pull request merged**